rules/no-typeof-undefined.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+const {findVariable} = require('eslint-utils');
+const {matches} = require('./selectors/index.js');
+
+const MESSAGE_ID = 'NO_TYPEOF_UNDEFINED';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `{{operator}}` operator instead of typeof with \'undefined\'.',
+};
+
+const selector = [
+	'BinaryExpression',
+	matches(['[right.value=undefined]', '[left.value=undefined]']),
```

Comment:
We need check it's string literal

```suggestion
matches(['[right.type="Literal"][right.value="undefined"]', '[left.type="Literal"][left.value="undefined"]']),
```